src/components/date-picker/DatePicker.tsx

```diff
@@ -442,24 +467,32 @@ class DatePicker extends React.Component<DatePickerProps> {
                                 text={error || ''}
                                 theme={TooltipTheme.ERROR}
                             >
-                                <input
-                                    ref={ref => {
-                                        this.dateInputEl = ref;
-                                    }}
-                                    className="date-picker-input"
-                                    disabled={isDisabled}
-                                    onBlur={this.handleInputBlur}
-                                    placeholder={placeholder || formatMessage(messages.chooseDate)}
-                                    required={isRequired}
-                                    type="text"
-                                    {...onChangeAttr}
-                                    onFocus={onFocus}
-                                    onKeyDown={this.handleInputKeyDown}
-                                    {...resinTargetAttr}
-                                    {...ariaAttrs}
-                                    {...inputProps}
-                                    {...valueAttr}
-                                />
+                                {customInput ? (
+                                    React.cloneElement(customInput, {
+                                        ref: (ref: HTMLInputElement) => {
+                                            this.dateInputEl = ref;
```

Comment:
Does the custom input field need to inherit the disabled/required states?

Reply:
Fixed!